src/hooks/useAllV3Ticks.ts

```diff
@@ -0,0 +1,108 @@
+import { Token } from '@uniswap/sdk-core'
+import { FeeAmount, TICK_SPACINGS } from '@uniswap/v3-sdk'
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk/dist/'
+import { ZERO_ADDRESS } from '../constants/misc'
+import { useEffect, useMemo, useState } from 'react'
+import { Result, useSingleCallResult, useSingleContractMultipleData } from 'state/multicall/hooks'
+import { useTickLens, useV3Factory } from './useContract'
+import { TickData } from 'constants/ticks'
+
+function bitmapIndex(tick: number, tickSpacing: number) {
+  return Math.floor(tick / tickSpacing / 256)
+}
+
+const REFRESH_FREQUENCY = { blocksPerFetch: 24 }
+
+// for now, reconsider using this function, it consumes a lot of data and cpu to fetch all the ticks.
+export function useAllV3Ticks(
+  token0: Token | undefined,
+  token1: Token | undefined,
+  feeAmount: FeeAmount | undefined
+): {
+  loading: boolean
+  syncing: boolean
+  error: boolean
+  valid: boolean
+  tickData: TickData[]
+} {
+  const tickSpacing = feeAmount && TICK_SPACINGS[feeAmount]
+
+  const minIndex = useMemo(
+    () => (tickSpacing ? bitmapIndex(nearestUsableTick(TickMath.MIN_TICK, tickSpacing), tickSpacing) : undefined),
+    [tickSpacing]
+  )
+  const maxIndex = useMemo(
+    () => (tickSpacing ? bitmapIndex(nearestUsableTick(TickMath.MAX_TICK, tickSpacing), tickSpacing) : undefined),
+    [tickSpacing]
+  )
+
+  const [tickDataLatestSynced, setTickDataLatestSynced] = useState<TickData[]>([])
+
+  // fetch the pool address
+  const factoryContract = useV3Factory()
+  const poolAddress = useSingleCallResult(factoryContract, 'getPool', [token0?.address, token1?.address, feeAmount])
+    .result?.[0]
```

Comment:
could look this up w/o network call

Reply:
how? I could use usePool() which will maybe have a cached response. Is there a more direct way?

Reply:
ah yeah, `Pool.getAddress` https://github.com/Uniswap/uniswap-v3-sdk/blob/main/src/entities/pool.ts